tools/nodejs/src/statement.cpp

```diff
@@ -166,7 +166,11 @@ static Napi::Value convert_col_val(Napi::Env &env, duckdb::Value dval, duckdb::L
 		value = Napi::Number::New(env, duckdb::DoubleValue::Get(dval));
 	} break;
 	case duckdb::LogicalTypeId::HUGEINT: {
-		value = Napi::Number::New(env, dval.GetValue<double>());
+		auto val = duckdb::HugeIntValue::Get(dval);
+		auto negative = val.upper < 0;
+		auto upper = negative ? val.upper * -1 : val.upper; // remove signing bit
```

Comment:
There's something subtly wrong with this long, causing the negative result to be out by two, but I'm not sure what it is

Reply:
Maybe we're missing this step that's done in ToString if the hugeint is negative:
```c++
	static void NegateInPlace(hugeint_t &input) {
		if (input.upper == NumericLimits<int64_t>::Minimum() && input.lower == 0) {
			throw OutOfRangeException("HUGEINT is out of range");
		}
		input.lower = NumericLimits<uint64_t>::Maximum() - input.lower + 1;
		input.upper = -1 - input.upper + (input.lower == 0);
	}
```

Reply:
That was it, cheers!